src/Common/src/Interop/Unix/System.Security.Cryptography.Native/Interop.EcDsa.ImportExport.cs

```diff
@@ -0,0 +1,153 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Runtime.InteropServices;
+using System.Security.Cryptography;
+using Microsoft.Win32.SafeHandles;
+using System;
+using System.Diagnostics;
+
+internal static partial class Interop
+{
+    internal static partial class Crypto
+    {
+        // todo: remove this
+        [DllImport(Libraries.CryptoNative, EntryPoint = "OutputDebugString")]
+        public static extern void OutputDebugString(string oidValue);
+
+        [DllImport(Libraries.CryptoNative, EntryPoint = "CryptoNative_EcKeyGetCurveType")]
+        internal static extern ECCurveType EcKeyGetCurveType(SafeEcKeyHandle key);
+
+        [DllImport(Libraries.CryptoNative, EntryPoint = "CryptoNative_EcKeyCreateByPublicKeyParameters")]
+        internal static extern SafeEcKeyHandle EcKeyCreateByPublicKeyParameters(
+            int nid, 
+            byte[] qx, int qxLength, 
+            byte[] qy, int qyLength, 
+            byte[] d, int dLength);
+
+        [DllImport(Libraries.CryptoNative, EntryPoint = "CryptoNative_EcKeyCreateByExplicitParameters")]
+        internal static extern SafeEcKeyHandle EcKeyCreateByExplicitParameters(
+            ECCurveType curveType,
+            byte[] qx, int qxLength,
+            byte[] qy, int qyLength,
+            byte[] d, int dLength,
+            byte[] p, int pLength,
+            byte[] a, int aLength,
+            byte[] b, int bLength,
+            byte[] gx, int gxLength,
+            byte[] gy, int gyLength,
+            byte[] n, int nLength,
+            byte[] h, int hLength,
+            byte[] s, int sLength);
+
+        [DllImport(Libraries.CryptoNative, EntryPoint = "CryptoNative_GetECKeyParameters")]
+        [return: MarshalAs(UnmanagedType.Bool)]
+        private static extern bool GetECKeyParameters(
+            SafeEcKeyHandle key, 
+            bool includePrivate,
+            out IntPtr qx_bn, out int x_cb,
+            out IntPtr qy_bn, out int y_cb,
+            out IntPtr d_bn_not_owned, out int d_cb);
+
+        internal static void GetECKeyParameters(SafeEcKeyHandle key, bool includePrivate, ref byte[] Qx, ref byte[] Qy, ref byte[] d)
+        {
+            if (key == null || key.IsInvalid)
+            {
+                Debug.Assert(false);
```

Comment:
Probably not necessary as we throw an exception on next line. I don't think we have any formal conventions for checking invariants like this?